src/Generator/Generators/CLI/CLIHeaders.cs

```diff
@@ -164,10 +164,7 @@ public void GenerateDeclContext(DeclarationContext decl)
             // Generate all the struct/class declarations for the module.
             foreach (var @class in decl.Classes)
             {
-                if (!@class.IsGenerated || @class.IsIncomplete || @class.IsDependent)
-                    continue;
-
-                if (@class.IsOpaque)
+                if (!@class.IsGenerated || (@class.IsIncomplete && !@class.IsOpaque) || @class.IsDependent)
```

Comment:
You only need to check IsOpaque, not IsIncomplete as well. Please fix here and anywhere else you do this.